### PR TITLE
mvt: init at 2.5.4

### PR DESCRIPTION
**Pull request by PapayaJackal**
Package [Mobile Verification Toolkit](https://mvt.re/) and its dependencies

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/mv/mvt/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    simplejson
+    tld
+  ];
+
+  meta = {
+    description = "Tool to facilitate the consensual forensic analysis of Android and iOS devices";
+    homepage = "https://docs.mvt.re/en/latest/";
+    license = lib.licenses.mvt;
+    changelog = "https://github.com/mvt-project/mvt/releases/tag/v${version}";
+    maintainers = with lib.maintainers; [ PapayaJackal ];
```
Comment (Aleksanaa) [inline pkgs/by-name/mv/mvt/package.nix:40]: missing `mainProgram`

File: pkgs/by-name/mv/mvt/package.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  pname = "mvt";
+  pyproject = true;
+  version = "2.5.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-u1OdhTrSfWSZrL4D+fRoa4c4xiS2jSJuF5T6E1jx3Q4=";
+  };
+
+  nativeBuildInputs = with python3.pkgs; [ setuptools ];
```
Comment (Aleksanaa) [inline pkgs/by-name/mv/mvt/package.nix:17]:
```suggestion
  build-system = with python3.pkgs; [ setuptools ];
```

File: pkgs/by-name/mv/mvt/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  propagatedBuildInputs = with python3.pkgs; [
```
Comment (Aleksanaa) [inline pkgs/by-name/mv/mvt/package.nix:19]:
```suggestion
  dependencies = with python3.pkgs; [
```

File: pkgs/development/python-modules/NSKeyedUnArchiver/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  lib,
+  pythonOlder,
+}:
+
+buildPythonPackage rec {
+  pname = "NSKeyedUnArchiver";
+  version = "1.2";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.8";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/NSKeyedUnArchiver/default.nix:13]:
```suggestion
```

No more python 3.8 in tree

File: pkgs/development/python-modules/iOSbackup/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  NSKeyedUnArchiver,
+  pycrypto,
+}:
+
+buildPythonPackage rec {
+  pname = "iOSbackup";
+  version = "0.9.925";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.7";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/iOSbackup/default.nix:15]:
```suggestion
```

File: pkgs/development/python-modules/iOSbackup/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-M1Rakknls/qq3x7ngv5r3823D64N77oazuM2pl+T0co=";
+  };
+
+  propagatedBuildInputs = [
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/iOSbackup/default.nix:22]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/NSKeyedUnArchiver/default.nix [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-OZWPGH8ggYYKxeg6CGyft6ZM9Yx/43z7YylzXlQcTVg=";
+  };
+
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/NSKeyedUnArchiver/default.nix:19]: Would be better to add `pythonImportsCheck` for both modules, see https://nixos.org/manual/nixpkgs/unstable/#using-pythonimportscheck

File: lib/licenses.nix [hunk 8]
```diff
@@ -936,6 +936,11 @@ lib.mapAttrs mkLicense ({
     fullName = "Microsoft Public License";
   };
 
+  mvt = {
+    fullName = "MVT License 1.1";
+    url = "https://docs.mvt.re/en/latest/license/";
+  };
```
Comment (alyssais) [inline lib/licenses.nix:942]: There's no need to add a licenses.nix entry for a bespoke, non-free license.  Just set it to `licenses.unfree`.

File: pkgs/by-name/mv/mvt/package.nix [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  nativeCheckInputs = with python3.pkgs; [
+    pytestCheckHook
+    pytest-mock
+    stix2
+  ];
+
+  meta = {
+    description = "Tool to facilitate the consensual forensic analysis of Android and iOS devices";
+    homepage = "https://docs.mvt.re/en/latest/";
+    license = lib.licenses.unfree;
```
Comment (fabaff) [inline pkgs/by-name/mv/mvt/package.nix:46]:
```suggestion
    # https://github.com/mvt-project/mvt/blob/main/LICENSE
    license = lib.licenses.unfree;
```

File: pkgs/development/python-modules/NSKeyedUnArchiver/default.nix [hunk 10]
```diff
@@ -0,0 +1,26 @@
+{
+  buildPythonPackage,
+  fetchPypi,
+  lib,
+  pythonOlder,
+}:
+
+buildPythonPackage rec {
+  pname = "NSKeyedUnArchiver";
```
Comment (fabaff) [inline pkgs/development/python-modules/NSKeyedUnArchiver/default.nix:9]: See https://nixos.org/manual/nixpkgs/stable/#contributing-guidelines

```suggestion
  pname = "nskeyedunarchiver";
```

File: pkgs/development/python-modules/NSKeyedUnArchiver/default.nix [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+  version = "1.2";
+  format = "setuptools";
```
Comment (fabaff) [inline pkgs/development/python-modules/NSKeyedUnArchiver/default.nix:11]:
```suggestion
  pyproject = true;
```

and add `build-system = [ setuptools ];`

File: pkgs/development/python-modules/iOSbackup/default.nix [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  pythonOlder,
+  NSKeyedUnArchiver,
+  pycrypto,
+}:
+
+buildPythonPackage rec {
+  pname = "iOSbackup";
```
Comment (fabaff) [inline pkgs/development/python-modules/iOSbackup/default.nix:11]: See comment above.

File: pkgs/development/python-modules/iOSbackup/default.nix [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+  version = "0.9.925";
+  format = "setuptools";
```
Comment (fabaff) [inline pkgs/development/python-modules/iOSbackup/default.nix:13]: See comment above.

File: pkgs/development/python-modules/iOSbackup/default.nix [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+    pycrypto
+    NSKeyedUnArchiver
+  ];
+
+  pythonImportsCheck = [ "iOSbackup" ];
+
+  meta = {
+    description = "Reads and extracts files from password-encrypted iOS backups";
+    homepage = "https://github.com/avibrazil/iOSbackup";
+    license = lib.licenses.lgpl2;
```
Comment (fabaff) [inline pkgs/development/python-modules/iOSbackup/default.nix:30]: There is no statement about `LGPLv2+` upstream, thus we should assume that it's `LGPL2`-only.

```suggestion
    license = lib.licenses.lgpl2Only;
```

**Pull request merged**